### DISCUSSION TITLE
Is this mod now obsolete with the game having implemented the function for AI Engineers to make passenger stops?

### DISCUSSION THREAD

**Discussion by Peccadilloz** [Erabior/RouteManager#103] [General]
Basically the title. Or does the mod offer more features than what was released by the game devs yesterday?

**Comment by Erabior** [comment]
While the core feature of the mod has been implemented into the game. There is definitely more to the mod than what is currently on the base game.

**Comment by IowaHawkeye13** [comment]
I do hope that you keep working on the mod.

I do like the game version in that if the game crashes, etc and you restart, there is not much hassle in getting the trains going again.  In your mod, this can be a hassle.

But I love how the mod will allow the trains to run all day.  Once they get to the last stop, they go in reverse.  I don't have to do anything!! YAY!!  I told Connor that they should implement your mod into the game.

I hope that you can get yours working soon.

Thanks!

**Comment by AdamC1228** [comment]
The mod is much more than the current game's implementation and more functionality is planned. There are a lot of cool features coming soon in v2.1 that are sure to interest many with special thanks to @amacro for all the contributions. 

Once I sort some things in my personal life out, you can expect some previews soon.

**Reply by TnNytehawk** [reply]
Excellent! Glad to hear that.  While the game now has some functionality for stopping at each station, I find myself really missing this mod's ability to keep the train going while I work on other tasks, and I'm really missing the threshold for stopping the train when water & coal is low.

Really missing this mod, and can't wait to see what comes with v2.1. :-)  Thanks!

**Reply by AdamC1228** [reply]
All of that continues to function regardless of the new features coming.

**Reply by TnNytehawk** [reply]
Dunno about everyone else, but for me and a couple other guys, RM2.0.0.8 kills the orders page after the last update, so we can't use the AI post last update at all. We had to remove the mod just to play with regular AI.  Since we all had the same results, we thought it was broken for everyone, but from what you just said maybe that's an isolated case?  Is RouteManager 2.0.0.8 working in Railroader 2024.2.x?

**Comment by AMacro** [comment]
> Dunno about everyone else, but for me and a couple other guys, RM2.0.0.8 kills the orders page after the last update, so we can't use the AI post last update at all. We had to remove the mod just to play with regular AI. Since we all had the same results, we thought it was broken for everyone, but from what you just said maybe that's an isolated case? Is RouteManager 2.0.0.8 working in Railroader 2024.2.x?

I can't say I've had an issue, what's it doing?

Might be worth raising an issue with some detail of what's happening.

**Reply by TnNytehawk** [reply]
Issue 105, Blank Orders Tab.  Raised by JohnWilkins82, but definitely what our group is seeing.

https://github.com/Erabior/RouteManager/issues/105
https://github.com/Erabior/RouteManager/discussions/104

**Comment by IowaHawkeye13** [comment]
Hey adam.

Would you be able to have route manager actually Implement a route? If I have two trains heading west from Sylva and I want the first one to go to Andrews and the one behind it to go to Alarka, can the second train throw its own switch at Alarka Junction?

**Reply by AMacro** [reply]
This is what I'm attempting to implement on the experimental UI.

Phase 1 a)
You can select the drop off points/stop points and which passengers you will carry overall e.g. you can pick up everyone at Sylva and drop anyone going to Ela and beyound at say, Whittier, before turning around to go back to Sylva, effectively running a transfer service.

Phase 1b)
Train looks at the route and determines the required positions of all the switches. As the train aproaches a switch it determines if it is set as per requirements, and if not, it determines if the current switch position will prevent the route from being complered. If the route can not be completed, the train stops and alerts you to throw the switch, otherwise it will plan a route through the switch.

A case where you may be able to plan a new route is if the train is set to enter the second passenger platform at say Whittier - either platform is acceptable. A case where you may not be able to is Alarka Jct, where you need to use the reversed track to get to Cochran.


Phase 2)
Allow the order of stations to be served to be set, rather than the default East/West (right/left by CTC).


Keep in mind that complex operations like this take time to implement and debug, especially when available time is limited (family/work/etc.).

**Comment by IowaHawkeye13** [comment]
Thanks Adam. I'm sure stuff like this takes a while. We are all looking forward to your updates.

**Reply by IowaHawkeye13** [reply]
Any updates Adam?
